apps/emqx_rule_engine/src/emqx_rule_funcs.erl

```diff
@@ -880,6 +882,18 @@ time_unit(<<"millisecond">>) -> millisecond;
 time_unit(<<"microsecond">>) -> microsecond;
 time_unit(<<"nanosecond">>) -> nanosecond.
 
+get_date(TimeUnit, Offset, FormatString) ->
```

Comment:
We also need to support formatting integer epoch to date strings, so I suggest changing the function definition of format_date/3 to:

```
format_date(Format, Offset, TimeUnit) -> xx;
format_date(Format, Offset, TimeUnit, Epoch) -> xx.
```